overlord/snapstate/autorefresh.go

```diff
@@ -745,15 +745,16 @@ func (e *timedBusySnapError) Is(err error) bool {
 	return ok
 }
 
-// inhibitRefresh returns an error if refresh is inhibited by running apps.
+// inhibitRefresh returns whether a refresh is forced due to inhibition
+// timeout or an error if refresh is inhibited by running apps.
 //
 // Internally the snap state is updated to remember when the inhibition first
 // took place. Apps can inhibit refreshes for up to "maxInhibition", beyond
 // that period the refresh will go ahead despite application activity.
-func inhibitRefresh(st *state.State, snapst *SnapState, snapsup *SnapSetup, info *snap.Info) error {
+func inhibitRefresh(st *state.State, snapst *SnapState, snapsup *SnapSetup, info *snap.Info) (inhibitionTimeout bool, err error) {
```

Comment:
are we expecting more values here? if so I'd add:

```
type inhibitionResult int
const (
    // not inhibited
    inhibitionNone inhibitionResult = iota
    // not inhibited as previous inhibition timed out
    inhibitionHoldExpired
)
```
though it still feels a bit odd

Reply:
This part would definitely benefit from a refactor. I think we could drop `timedBusySnapError` later because emitting notification from snapd will be eventually replaced by notices (with a fallback to warnings) so they might need to stay until we figure the replacement out.

The refactor will need careful thought to avoid breaking things, but definitely important to do eventually.